### ISSUE TITLE
Anomaly "File "printing/ppconstr.ml", line 398, characters 14-20: Assertion failed."

### ISSUE THREAD

**Issue by Zimmi48** [coq/coq#7867]
#### Version

8.8.0 and master

#### Description of the problem

```
Class class := { foo : nat }.
Instance : class := { foo := 0 ; bar := 0 }.
```

gives:

```
Error: Anomaly "File "printing/ppconstr.ml", line 398, characters 14-20: Assertion failed."
Please report at http://coq.inria.fr/bugs/.
```
The line number is for current master, i.e. this is:

https://github.com/coq/coq/blob/6715e6801c1d285a12eeca55dd8b831d7efb8c0d/printing/ppconstr.ml#L398

**Comment by yangky11** [comment]
I encountered the same issue.

**Comment by herbelin** [comment]
Fixed by #8286. The expected error message is however slightly difficult to decipher. One gets:
```
Error: Mismatched contexts while declaring instance:  
Expected: (foo : nat)
 Found:   0  0
```
Why not to rather say `The field "bar" does not exist.` as we have when building a record with an unknown field (`Check {| foo := 0 ; bar := 0 |}.`?
